File: sdk/src/main/java/com/cxense/cxensesdk/model/WidgetRequest.java
package com.cxense.cxensesdk.model;

/**
 * Request widget data object for server
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */

public final class WidgetRequest {
    public String widgetId;
    public ContentUser user;
    public WidgetContext context;
    public String con;

    public WidgetRequest() {
    }

    public WidgetRequest(String widgetId, WidgetContext widgetContext, ContentUser widgetUser, String consent) {
        this();
        this.widgetId = widgetId;
        user = widgetUser;
        context = widgetContext;
        con = consent;
    }
}

File: sdk/src/main/java/io/piano/android/cxense/model/WidgetRequest.kt
package io.piano.android.cxense.model

import com.squareup.moshi.JsonClass

/**
 * Request widget data object for server
 */
@JsonClass(generateAdapter = true)
public class WidgetRequest(
    public val widgetId: String,
    public val consent: List<String>,
    public val context: WidgetContext? = null,
    public val user: ContentUser? = null,
    public val tag: String? = null,
    public val prnd: String? = null,
    public val experienceId: String? = null,
)
